### PR TITLE
[Validator] Add "Translatable Objects" example

### LINKED ISSUE #19049: [validation/translations] Add `TranslatableMessage` class to doc
Can we add a paragraph in https://github.com/symfony/symfony-docs/blob/6.4/validation/translations.rst about how to translate with [TranslatableMessage](https://github.com/symfony/symfony/blob/6.4/src/Symfony/Component/Translation/TranslatableMessage.php) since it is out from [Symfony 6.2 (blog)](https://symfony.com/blog/new-in-symfony-6-2-better-php-translation-extractor)

Like so:
```php

use Symfony\Component\Translation\TranslatableMessage;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Component\Validator\Context\ExecutionContextInterface;

#[Assert\Callback]
public function validate(ExecutionContextInterface $context, mixed $payload): void
{
    $context->buildViolation(new TranslatableMessage('author.name.not_blank', [], 'validators'))
            ->atPath('name')
            ->addViolation()
    ;
}
```

### PR DESCRIPTION
**Pull request by cavasinf**
Resolve #19049

### CODE REVIEW

File: validation/translations.rst [hunk 1]
```diff
@@ -123,6 +123,40 @@ Now, create a ``validators`` catalog file in the ``translations/`` directory:
...
+            // check if the name is actually a fake name
+            if (in_array($this->getFirstName(), $fakeNames)) {
+                $context->buildViolation(new TranslatableMessage('author.name.fake', [], 'validators'))
+                    ->atPath('firstName')
+                    ->addViolation()
+                ;
+            }
+        }
+
+Learn more about
```
Comment (cavasinf) [inline validation/translations.rst]:
```suggestion
Learn more about
```

**Comment by alexandre-daubois** [comment]
First of all, congrats on your first doc contribution!

I think we can make this even simpler. Just after "You may need to clear your cache (even in the dev environment) after creating this file for the first time.", we could add something like:

```rst
You can also use :class:`Ns\\TranslatableMessage` to build your violation message::

// Your PHP example

You can learn more about translatable messages in :ref:`the dedicated section <translatable-objects>`.
```

If you can, try not to get further than 80 chars lines 🙂

**Comment by cavasinf** [comment]
Sure 👍 
I made it shorter as you recommended.

I still have the code blocks error about an undefined class. 
Looking at https://github.com/symfony/symfony-docs/blob/6.4/reference/constraints/NotBlank.rst it is made the same way, am I missing something?

File: validation/translations.rst [hunk 2]
```diff
@@ -123,6 +123,29 @@ Now, create a ``validators`` catalog file in the ``translations/`` directory:
 You may need to clear your cache (even in the dev environment) after creating
 this file for the first time.
 
+You can also use :class:`Ns\\TranslatableMessage` to build your violation message::
```
Comment (alexandre-daubois) [inline validation/translations.rst]:
```suggestion
You can also use :class:`Symfony\\Component\\Translation\\TranslatableMessage` to build your violation message::
```

`Ns` in my example was just a placeholder for the real namespace :slightly_smiling_face:

**Comment by alexandre-daubois** [comment]
I think it's safe to ignore the error of the codeblocks as we're importing a "folder" here (and the parser is surely looking for some class or something)

File: validation/translations.rst (hunk 2 above)
Comment (cavasinf) [reply validation/translations.rst]: Ok
I though you gave a shortcut or something

**Comment by cavasinf** [comment]
> I think it's safe to ignore the error of the codeblocks as we're importing a "folder" here (and the parser is surely looking for some class or something)

Good to go then 🤔

File: validation/translations.rst [hunk 3]
```diff
@@ -123,6 +123,29 @@ Now, create a ``validators`` catalog file in the ``translations/`` directory:
...
+    use Symfony\Component\Validator\Context\ExecutionContextInterface;
+    
+    #[Assert\Callback]
+    public function validate(ExecutionContextInterface $context, mixed $payload): void
+    {
+        // somehow you have an array of "fake names"
+        $fakeNames = [/* ... */];
+    
+        // check if the name is actually a fake name
+        if (in_array($this->getFirstName(), $fakeNames)) {
```
Comment (alexandre-daubois) [inline validation/translations.rst]:
```suggestion
        if (in_array($this->getFirstName(), $fakeNames, true)) {
```

Let's be strict where we can :smile:

**Review by alexandre-daubois — APPROVED** [review]
Just a little nitpick, otherwise it's fine to me! :slightly_smiling_face:

**Comment by javiereguiluz** [comment]
Nice update! Florian, thanks for adding this and for your patience during the review process (thanks to Alex for reviewing it too).

And congrats on your first Symfony Docs contribution 🎉

**Comment by SebTM** [comment]
Hey, not sure about this - stumbled on the docs and for v7 and at least v6 I'm using atm. 

https://github.com/symfony/validator/tree/7.0/Context#L157
https://github.com/symfony/validator/blob/6.4/Context/ExecutionContext.php#L157

this seems not working and triggers an error now as it's typed to string and has translation built-in ✌🏻

**Comment by xabbuh** [comment]
@SebTM Please open a new issue if you think that you discovered a bug. Comments on merged PRs are likely to get lost.

**Pull request merged**